Unresolved import `core::sync::atomic::AtomicU64`

I am trying to use this library with a `ESP32-C3` unfortunately when adding `esp32-nimble` to the dependencies I get a compiler error:
```
error[E0432]: unresolved import `core::sync::atomic::AtomicU64`
   --> /home/derfetzer/.cargo/registry/src/index.crates.io-6f17d22bba15001f/atomic-polyfill-1.0.3/src/polyfill.rs:6:17
    |
6   |         pub use core::sync::atomic::$atomic_type;
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `AtomicU64` in `sync::atomic`
...
169 | atomic_int!(u64, AtomicU64, polyfill_u64);
    | -----------------------------------------
    | |                |
    | |                help: a similar name exists in the module: `AtomicU32`
    | in this macro invocation
    |
    = note: this error originates in the macro `atomic_int` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0432]: unresolved import `core::sync::atomic::AtomicI64`
   --> /home/derfetzer/.cargo/registry/src/index.crates.io-6f17d22bba15001f/atomic-polyfill-1.0.3/src/polyfill.rs:6:17
    |
6   |         pub use core::sync::atomic::$atomic_type;
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `AtomicI64` in `sync::atomic`
...
174 | atomic_int!(i64, AtomicI64, polyfill_i64);
    | -----------------------------------------
    | |                |
    | |                help: a similar name exists in the module: `AtomicI32`
    | in this macro invocation
    |
    = note: this error originates in the macro `atomic_int` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0432`.
error: could not compile `atomic-polyfill` (lib) due to 2 previous errors

```

I am using the `esp-idf-template` and only added the `CONFIG_BT_*` flags and this library as a dependency.

It seems there have been issues with `Xtensa` targets in the past (https://github.com/embassy-rs/atomic-polyfill/issues/14).
The last hint there was to update the compiler version but I did `espup update` just today.

Do you have any idea what the problem could be?

Thanks in advance!

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,8 +26,8 @@ opt-level = "z"
 [features]
 default = ["std"]
 
-std = ["esp-idf-svc/std", "esp-idf-hal/std", "critical-section/std"]
-no_std = ["esp-idf-hal/critical-section"]
+std = ["esp-idf-svc/std", "esp-idf-hal/std", "once_cell/std"]
+no_std = ["once_cell/critical-section", "esp-idf-hal/critical-section"]
 
 [dependencies]
 log = { version = "0.4", default-features = false }
@@ -37,9 +37,8 @@ esp-idf-svc = { version = "0.47", default-features = false, features = ["alloc"]
 embedded-svc = { version = "0.26", default-features = false }
 
 bitflags = { version = "2.4.1" }
-critical-section = { version = "1.1.1" }
 embassy-sync = { version = "0.3.0" }
-once_cell = { version = "1.16.0", default-features = false, features = ["critical-section"] }
+once_cell = { version = "1.16.0", default-features = false }
 uuid = { version = "1.2.2", default-features = false, features = ["macro-diagnostics"] }
 
 [dev-dependencies]

